src/LiveAudioVisualizer/LiveAudioVisualizer.tsx

```diff
@@ -131,6 +131,12 @@ const LiveAudioVisualizer: (props: Props) => ReactElement = ({
     }
   }, [analyser, context.state]);
 
+  useEffect(() => {
+    if (context.state !== "closed") {
+      context.close();
+    }
+  }, []);
+
```

Comment:
Apologies for getting to this late.

But wouldn't the if condition be more suitable in a clean up function?

Reply:
```suggestion
  useEffect(() => {
    return () => {
      if (context.state !== "closed") {
        context.close();
      }
    }
  }, []);

```

Reply:
oof, my bad